src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -21,22 +22,32 @@ import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.OffenderBooking
 
 
 @Service
-class PrisonerIndexService(val nomisService: NomisService,
-                           val prisonerARepository: PrisonerARepository,
-                           val prisonerBRepository: PrisonerBRepository,
-                           val indexQueueService : IndexQueueService,
-                           val indexStatusService: IndexStatusService,
-                           val searchClient: SearchClient,
-                           @Value("\${index.page.size:1000}") val pageSize : Int
+class PrisonerIndexService(private val nomisService: NomisService,
+                           private val prisonerARepository: PrisonerARepository,
+                           private val prisonerBRepository: PrisonerBRepository,
+                           private val indexQueueService : IndexQueueService,
+                           private val indexStatusService: IndexStatusService,
+                           private val searchClient: SearchClient,
+                           private val telemetryClient: TelemetryClient,
+                           @Value("\${index.page.size:1000}") private val pageSize : Int
 ) {
     companion object {
         val log: Logger = LoggerFactory.getLogger(this::class.java)
     }
 
     fun indexPrisoner(prisonerId: String) : Prisoner? {
-        return nomisService.getOffender(prisonerId)?.let {
+        val prisoner = nomisService.getOffender(prisonerId)?.let {
             sync(it)
         }
+
+        if (prisoner == null) {
```

Comment:
you can use something like `:?  telemetryClient.trackEvent(` rather than if

Reply:
Yeah I kinda figured there would be a nicer way of doing this....first time doing 'K'